src/main/jbake/content/community/docs/developer-guide/alerts.adoc

```diff
@@ -389,3 +451,420 @@ public class Action {
+
+This expression will group events filtered by a expression
+
+[source]
+filter(
+    (category == "Credit Check" && text == "Exceptionally Good") ||
+    (category == "Stock Check" && text == "Out of Stock")
+)
+
+Note that this expression doesn't define an explicit sliding time window, so it will use a default expiration window.
+
+Additional details can be consulted on the JavaDoc of the implementation and examples:
+
+* link:https://github.com/hawkular/hawkular-alerts/blob/master/hawkular-alerts-engine-extensions/hawkular-alerts-events-aggregation/src/main/java/org/hawkular/alerts/extensions/Expression.java[Events Aggregation Expression JavaDoc]
+* link:https://github.com/hawkular/hawkular-alerts/tree/master/examples/events-aggregation[Events Aggregation Expression Examples]
+
+== Elasticsearch Integration
+
+=== Elasticsearch Alerter
+
+The Elasticsearch Alerter will listen for triggers tagged with "Elasticsearch" tag. The Alerter will schedule a
+periodic query to an Elasticsearch system with the info provided from the tagged trigger context. The Alerter
+will convert Elasticsearch documents into Hawkular Alerting Events and send them into the Alerting engine.
+
+The Elasticsearch Alerter uses the following conventions for trigger tags and context:
+
+[cols="^2,10"]
+|=======================
+| *Required* |
+_trigger.tags["Elasticsearch"] = "<any description>"_ +
+ +
+An _Elasticsearch_ tag is required for the alerter to detect this trigger will query to an Elasticsearch system. +
+Value is not necessary, it can be used as a description, it is reserved for future uses. +
+ +
+i.e.
+ +
+trigger.tags["Elasticsearch"] = ""                          // Empty value is valid +
+trigger.tags["Elasticsearch"] = "OpenShift Logging System"  // It can be used as description
+| *Required* |
+_trigger.context["timestamp"] = "<timestamp field>"_ +
+ +
+Documents fetched from Elasticsearch need a date field to indicate the timestamp. +
+This timestamp will be used in the queries to fetch documents in interval basis. +
+ +
+If there is not defined a specific pattern under the trigger.context["timestamp.pattern"] it will follow the default patterns: +
+ +
+"yyyy-MM-dd'T'HH:mm:ss.SSSSSSZ" +
+"yyyy-MM-dd'T'HH:mm:ssZ"
+| *Required* |
+_trigger.context["mapping"] = "<mapping_expression>"_ +
+ +
+A mapping expression defines how to convert an Elasticsearch document into a Hawkular Event. +
+ +
+Mapping expression syntax (BNF): +
+ +
+<mapping_expression> ::= <mapping> \| <mapping> "," <mapping_expression> +
+<mapping> ::= <elasticsearch_field> [ "\|" "'" <DEFAULT_VALUE> "'" ] ":" <hawkular_event_field>
+<elasticsearch_field> ::= "index" \| "id" \| <SOURCE_FIELD> +
+<hawkular_event_field> ::= "id" \| "ctime" \| "dataSource" \| "dataId" \| "category" \| "text" \| "context" \| "tags" +
```

Comment:
Not as difficult to parse as the above DSL, but again, maybe easier if add some blanklines in between the entries.